test/build_processed_data_test.py

```diff
@@ -1,6 +1,57 @@
-from libs import build_processed_dataset
+from unittest.mock import patch
 
+import pytest
 
-def test_get_testing_df():
-    testing_df = build_processed_dataset.get_testing_timeseries_by_state("MA")
-    assert testing_df is not None
+from libs import build_processed_dataset, validate_results
+from libs.datasets import CommonFields, CDSDataset, CovidTrackingDataSource
+import pandas as pd
+from libs.datasets.results_schema import CUMULATIVE_POSITIVE_TESTS, CUMULATIVE_NEGATIVE_TESTS
+
+from libs.enums import Intervention
+from more_itertools import one
+
+
+def test_get_usa_by_states_df():
+    empty_df_with_state = pd.DataFrame([], columns=["state",])
+    with patch(
+        "libs.build_processed_dataset.get_state_projections_df", return_value=empty_df_with_state
+    ):
+        df = build_processed_dataset.get_usa_by_states_df(
+            input_dir="/tmp", intervention_type=Intervention.OBSERVED_INTERVENTION.value
+        )
+    validate_results.validate_states_df("TX", df)
+
+    tx_record = one(df.loc[df[CommonFields.STATE_FULL_NAME] == "Texas"].to_dict(orient="records"))
+    assert tx_record[CUMULATIVE_POSITIVE_TESTS] > 100
+    assert tx_record[CUMULATIVE_NEGATIVE_TESTS] > 100
+
+
+def test_get_testing_df_by_state():
+    positive_field = CovidTrackingDataSource.Fields.POSITIVE_TESTS
+    negative_field = CovidTrackingDataSource.Fields.NEGATIVE_TESTS
+    df = build_processed_dataset.get_testing_timeseries_by_state("MA")
+    assert positive_field in df.columns
+    assert negative_field in df.columns
+    # TODO(tom): have build_processed_dataset return a real datetime dtype so callers don't need to parse a string.
+    df[CommonFields.DATE] = pd.to_datetime(df[CommonFields.DATE], format="%m/%d/%y")
+    df.set_index(CommonFields.DATE, inplace=True)
+    # No joke, our only source of state-level timeseries testing data is CDSDataset and it has NaN for MA until April 6.
+    assert 0 < df.at["2020-04-10", positive_field] < df.at["2020-05-01", positive_field]
+    assert 0 < df.at["2020-04-10", negative_field] < df.at["2020-05-01", negative_field]
+
+
+# Check San Francisco/06075 and Houston (Harris County, TX)/48201
+@pytest.mark.parametrize(
+    "fips", ["06075", "48201"],
+)
+def test_get_testing_by_fips(fips):
+    df = build_processed_dataset.get_testing_timeseries_by_fips(fips)
```

Comment:
I know this uses the combined datasets under the hood now, but do you think we should explicitly check combined dataset county test data to make sure we have at least some values?

Reply:
Good idea. I added a check of cases and deaths. there isn't much data about tests at the county level.